api/yelp_beans/logic/metrics.py

```diff
@@ -1,28 +1,28 @@
 from collections import defaultdict
 
+from sqlalchemy.orm import joinedload
 from yelp_beans.logic.meeting_spec import get_meeting_datetime
 from yelp_beans.logic.meeting_spec import get_specs_for_current_week
+from yelp_beans.logic.meeting_spec import get_specs_for_current_week_query
 from yelp_beans.logic.meeting_spec import get_users_from_spec
 from yelp_beans.models import Meeting
 from yelp_beans.models import MeetingParticipant
 from yelp_beans.models import MeetingRequest
+from yelp_beans.models import MeetingSpec
 from yelp_beans.models import MeetingSubscription
-from yelp_beans.models import User
+from yelp_beans.models import UserSubscriptionPreferences
 
 
 def get_subscribers():
-    users = User.query.all()
+    sub_prefs = UserSubscriptionPreferences.query.options(joinedload(UserSubscriptionPreferences.user)).all()
     subscriptions = MeetingSubscription.query.all()
 
-    metrics = defaultdict(set)
     # creates metrics keys for all subscriptions including ones without users
-    for subscription in subscriptions:
-        metrics[subscription.id] = []
+    metrics = {subscription.id: [] for subscription in subscriptions}
 
     # creates metrics keys for all subscriptions that have users with user data
-    for user in users:
-        for preference in user.subscription_preferences:
-            metrics[preference.subscription_id].append(user.email)
+    for preference in sub_prefs:
+        metrics[preference.subscription_id].append(preference.user.email)
```

Comment:
I may have bad data (locally due to deleting a subscription I created) but testing this, it fails for case where preference.subscription_id is None.  May be you want to handle this case since there's no way to mark a subscription inactive so deleting may be valid. If we don't think this is ever a case to consider, feel free to ignore

Reply:
Added `subscription_id.isnot(None)` to the query to filter out these rows.